modules/social_features/social_event/src/Controller/EnrollRequestDialogController.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Drupal\social_event\Controller;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Ajax\AjaxResponse;
+use Drupal\Core\Ajax\OpenModalDialogCommand;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Form\FormBuilder;
+use Drupal\Core\Url;
+use Drupal\node\NodeInterface;
+
+/**
+ * Contains methods for the modal form when requesting to enroll in an event.
+ *
+ * @package Drupal\social_event\Controller
+ */
+class EnrollRequestDialogController extends ControllerBase {
+
+  /**
+   * The form builder.
+   *
+   * @var \Drupal\Core\Form\FormBuilder
+   */
+  protected $formBuilder;
+
+  /**
+   * The ModalFormExampleController constructor.
+   *
+   * @param \Drupal\Core\Form\FormBuilder $formBuilder
+   *   The form builder.
+   */
+  public function __construct(FormBuilder $formBuilder) {
+    $this->formBuilder = $formBuilder;
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   The Drupal service container.
+   *
+   * @return static
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('form_builder')
+    );
+  }
+
+  /**
+   * Helper method so we can have consistent dialog options.
+   *
+   * @return string[]
+   *   An array of jQuery UI elements to pass on to our dialog form.
+   */
+  protected static function getDataDialogOptions() {
+    return [
+      'dialogClass' => 'form--default',
+      'closeOnEscape' => TRUE,
+      'width' => '400',
```

Comment:
We should see if we can make this wider (as per design) and also have it scale nicely on mobile. Ask Alexander if needed.